src/Idris/Elab/Utils.hs

```diff
@@ -62,19 +62,19 @@ checkDeprecated fc n
     = do r <- getDeprecated n
          case r of
               Nothing -> return ()
-              Just r -> do iWarn fc $ text "Use of deprecated name " <> annName n
+              Just r -> iWarn fc $ text "Use of deprecated name "
+                                 <> annName n
                                  <> case r of
                                          "" -> Util.Pretty.empty
-                                         _ -> line <> text r
+                                         _  -> line <> text r
 
 
 checkFragile :: FC -> Name -> Idris ()
 checkFragile fc n = do
   r <- getFragile n
   case r of
     Nothing -> return ()
-    Just r  -> do
-      iWarn fc $ text "Use of a fragile construct "
+    Just r  -> iWarn fc $ text "Use of a fragile construct "
                      <> annName n
```

Comment:
This should probably move to the right, because its parent expression has moved to the right. Alternatively, the dangling "do" could be removed without moving the code up to the previous line.